examples/cpp/filter-data.cpp

```diff
@@ -0,0 +1,108 @@
+#include <catch2/catch_test_macros.hpp>
+#include <string>
+#include "testHelpers.hpp"
+#include <cpprealm/sdk.hpp>
+
+// :snippet-start: models
+struct Item : realm::object<Item> {
+    realm::persisted<std::string> name;
+    realm::persisted<bool> isComplete;
+    realm::persisted<std::optional<std::string>> assignee;
+    realm::persisted<int64_t> priority;
+    realm::persisted<int64_t> progressMinutes;
+
+    static constexpr auto schema = realm::schema("Item",
+        realm::property<&Item::name>("name"),
+        realm::property<&Item::isComplete>("isComplete"),
+        realm::property<&Item::assignee>("assignee"),
+        realm::property<&Item::priority>("priority"),
+        realm::property<&Item::progressMinutes>("progressMinutes"));
+};
+
+struct Project : realm::object<Project> {
+    realm::persisted<std::string> name;
+    realm::persisted<std::vector<Item>> items;
+
+    static constexpr auto schema = realm::schema("Project",
+        realm::property<&Project::name>("name"),
+        realm::property<&Project::items>("items"));
+};
+// :snippet-end:
+
+TEST_CASE("set up tests", "[write]") {
+    // :snippet-start: set-up-filter-data-tests
+    auto realm = realm::open<Project, Item>();
+
+    auto item1 = Item {
+        .name = "Save the cheerleader",
+        .assignee = std::string("Peter"),
+        .isComplete = false,
+        .priority = 6,
+        .progressMinutes = 30
+    };
+
+    auto project = Project {
+        .name = "New project"
+    };
+
+    project.items.push_back(item1);
+
+    realm.write([&project, &realm] {
+        realm.add(project);
+    });
+
+    auto items = realm.objects<Item>();
+    auto projects = realm.objects<Project>();
+    // :snippet-end:
+
+    CHECK(project.is_managed());
+    CHECK(!item1.is_managed());
+
+    CHECK(items.size() >= 1);
+    CHECK(projects.size() >= 1);
+
+    // :snippet-start: comparison-operators
+    auto highPriorityItems = items.where([](auto &item) {
```

Comment:
Minor, but where the item is immutable we should mark it const:
```suggestion
    auto highPriorityItems = items.where([](auto const& item) {
```